pyintegration/deephaven2/dtypes.py

```diff
@@ -99,31 +108,236 @@ def array_from(self, values: Iterable):
+            else:
+                return self.array_of(np.asarray(seq))
+        except Exception as e:
+            raise DHError(e, f"failed to create a Java {self.j_name} array.") from e
+
+
+# region convenience enum and module level type aliases
+class DTypes(Enum):
+    """ An enum that defines a predefined set of supported data types in Deephaven. """
+
+    bool_ = DType(j_name="java.lang.Boolean", qst_type=_JQstType.booleanType(), )
+    byte = IntegerDType(j_name="byte", qst_type=_JQstType.byteType(), np_dtypes=(np.int8, np.uint8),
+                        null_value=NULL_BYTE)
+    int8 = byte
+    short = IntegerDType(j_name="short", qst_type=_JQstType.shortType(), np_dtypes=(np.int16, np.uint16),
+                         null_value=NULL_SHORT)
+    int16 = short
+    char = CharDType(j_name="char", qst_type=_JQstType.charType(), np_dtypes=(), null_value=NULL_CHAR)
+    int_ = IntegerDType(j_name="int", qst_type=_JQstType.intType(), np_dtypes=(np.int32, np.uint32),
+                        null_value=NULL_INT)
+    int32 = int_
+    long = IntegerDType(j_name="long", qst_type=_JQstType.longType(), np_dtypes=(np.int64, np.uint64),
+                        null_value=NULL_LONG)
+    int64 = long
+    float_ = FloatingDType(j_name="float", qst_type=_JQstType.floatType(), np_dtypes=(np.float32,),
+                           null_value=NULL_FLOAT)
+    single = float_
+    float32 = float_
+    double = FloatingDType(j_name="double", qst_type=_JQstType.doubleType(), np_dtypes=(np.float64,),
+                           null_value=NULL_DOUBLE)
+    float64 = double
+    string = DType(j_name="java.lang.String", qst_type=_JQstType.stringType())
+    BigDecimal = DType(j_name="java.math.BigDecimal")
+    StringSet = DType(j_name="io.deephaven.stringset.StringSet")
+    DateTime = DType(j_name="io.deephaven.time.DateTime")
+    Period = DType(j_name="io.deephaven.time.Period")
+    PyObject = DType(j_name="org.jpy.PyObject")
+    JObject = DType(j_name="java.lang.Object")
+
+
+bool_ = DTypes.bool_.value
+byte = DTypes.byte.value
+int8 = DTypes.int8.value
+short = DTypes.short.value
+int16 = DTypes.int16.value
+char = DTypes.char.value
+int_ = DTypes.int_.value
+int32 = DTypes.int32.value
+long = DTypes.long.value
+int64 = DTypes.int64.value
+float_ = DTypes.float_.value
+single = DTypes.single.value
+float32 = DTypes.float32.value
+double = DTypes.double.value
+float64 = DTypes.float64.value
+string = DTypes.string.value
+BigDecimal = DTypes.BigDecimal.value
+StringSet = DTypes.StringSet.value
+DateTime = DTypes.DateTime.value
+Period = DTypes.Period.value
+PyObject = DTypes.PyObject.value
+JObject = DTypes.JObject.value
+
+
+# endregion
+
+# region helper functions
+
+def j_class_lookup(j_class: Any) -> DType:
+    if not j_class:
+        return None
+
+    j_type = jpy.get_type(j_class.getName())
+    for t in DTypes:
+        if t.value.j_type == j_type:
+            return t.value
 
 
 def j_array_list(values: Iterable):
```

Comment:
This line raises a question.  In the methods above, `Sequence` is used.  Since `Sequence` is an `Iterable`, should the prior methods use `Iterable` instead of `Sequence` in order to be more general, or is there a reason to keep `Sequence`?

Reply:
JPY requires a finite Sequence to create a Java array.